packages/http-server-js/src/scripts/scaffold/bin.mts

```diff
@@ -664,10 +662,26 @@ function* emitControllerOperationHandlers(
       yield `async ${opName}(ctx: HttpContext, ${paramsDeclarationLine}): ${returnType} {`;
     }
 
-    yield "  throw new NotImplementedError();";
+    const mockReturn = mockType(op.returnType);
+
+    if (mockReturn === undefined) {
+      importNotImplementedError = true;
+      yield "  throw new NotImplementedError();";
+    } else if (mockReturn === "void") {
```

Comment:
This seems unreachable because the case for TypeSpec void maps to JS `undefined`.

Reply:
good catch! thanks :)